easybuild/framework/easyblock.py

```diff
@@ -1390,6 +1390,37 @@ def make_module_description(self):
         """
         return self.module_generator.get_description()
 
+    def make_module_pythonpath(self):
+        """
+        Add lines for module file to update $PYTHONPATH or $EBPYTHONPREFIXES,
+        if they aren't already present and the standard lib/python*/site-packages subdirectory exists
+        """
+        lines = []
+        if not os.path.isfile(os.path.join(self.installdir, 'bin', 'python')):  # only needed when not a python install
+            python_subdir_pattern = os.path.join(self.installdir, 'lib', 'python*', 'site-packages')
+            candidate_paths = (os.path.relpath(path, self.installdir) for path in glob.glob(python_subdir_pattern))
+            python_paths = [path for path in candidate_paths if re.match(r'lib/python\d+\.\d+/site-packages', path)]
+
+            runtime_deps = [dep['name'] for dep in self.cfg.dependencies(runtime_only=True)]
+            use_ebpythonprefixes = all([
```

Comment:
We also have to take into account `multi_deps` here, if that's being used we really need to use `$EBPYTHONPREFIXES` instead of `$PYTHONPATH`...

Reply:
Normal scenarios would be
1. Using Python, multiple paths -> use ebpythonprefixes
2. Using Python, preferring EBPYTHONPREFIXES and not forcing pythonpath -> use ebpythonprefixes
3. Not using Python -> use PYTHONPATH
so maybe the logic should be 
```
            use_ebpythonprefixes = 'Python' in runtime_deps and (len(python_paths) > 1 or (build_option('prefer_python_search_path') == EBPYTHONPREFIXES,
                not self.cfg['force_pythonpath']))
```

Reply:
I broke that down and added logging for it too in https://github.com/Micket/easybuild-framework/pull/6, and enhanced the test to also check the `multi_deps` case